File: server/Messages/Event.cs
using System;
using System.Net;
using System.Xml;
using System.Xml.Serialization;

namespace RefBox
{
	[Serializable, XmlRoot(ElementName = "event", Namespace = "")]
	public class Event
	{
		/// <summary>
		/// Gets or sets the type of the signal
		/// </summary>
		[XmlAttribute("type")]
		public string Type { get; set; }

		/// <summary>
		/// Gets the signal value.
		/// </summary>
		[XmlText]
		public string Value { get; set; }

		/// <summary>
		/// Gets or sets the IPEndPoint where message comes from.
		/// </summary>
		[XmlIgnore]
		public IPEndPoint Source{ get; set; }
	}
}

File: refbox_rosclient/src/Event.cpp
#include "Event.h"
using namespace refbox;

Event::Event(){}

Event::Event(const Event& other):
	type(other.type), value(other.value), source(other.source){}

Event::Event(const std::string& type, const std::string& value):
	type(type), value(value){}

Event::Event(const std::string& type, const std::string& value, const EndPoint& source):
	type(type), value(value), source(source){}